aws/src/integration/java/org/apache/iceberg/aws/TestKeyManagementClient.java

```diff
@@ -0,0 +1,125 @@
+import software.amazon.awssdk.services.kms.model.CreateKeyResponse;
+import software.amazon.awssdk.services.kms.model.DataKeySpec;
+import software.amazon.awssdk.services.kms.model.KeySpec;
+import software.amazon.awssdk.services.kms.model.ScheduleKeyDeletionRequest;
+import software.amazon.awssdk.services.kms.model.ScheduleKeyDeletionResponse;
+
+@EnabledIfEnvironmentVariables({
+  @EnabledIfEnvironmentVariable(named = AwsIntegTestUtil.AWS_ACCESS_KEY_ID, matches = ".*"),
+  @EnabledIfEnvironmentVariable(named = AwsIntegTestUtil.AWS_SECRET_ACCESS_KEY, matches = ".*"),
+  @EnabledIfEnvironmentVariable(named = AwsIntegTestUtil.AWS_SESSION_TOKEN, matches = ".*"),
+  @EnabledIfEnvironmentVariable(named = AwsIntegTestUtil.AWS_TEST_ACCOUNT_ID, matches = "\\d{12}")
+})
+public class TestKeyManagementClient {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TestKeyManagementClient.class);
+
+  private static KmsClient kmsClient;
+  private static String keyId;
+
+  @BeforeAll
+  public static void beforeClass() {
+    kmsClient = AwsClientFactories.defaultFactory().kms();
+    CreateKeyRequest createKeyRequest =
+        CreateKeyRequest.builder()
+            .keySpec(KeySpec.SYMMETRIC_DEFAULT)
+            .description(
+                "Iceberg integration test key for " + TestKeyManagementClient.class.getName())
+            .build();
+    CreateKeyResponse response = kmsClient.createKey(createKeyRequest);
+    keyId = response.keyMetadata().keyId();
+  }
+
+  @Test
+  public void testKeyWrapping() {
+    AwsKeyManagementClient keyManagementClient = new AwsKeyManagementClient();
+    try {
+      keyManagementClient.initialize(ImmutableMap.of());
+
+      ByteBuffer key = ByteBuffer.wrap(new String("super-secret-table-master-key").getBytes());
+      ByteBuffer encryptedKey = keyManagementClient.wrapKey(key, keyId);
+
+      assertThat(keyManagementClient.unwrapKey(encryptedKey, keyId)).isEqualTo(key);
+    } finally {
+      keyManagementClient.close();
+    }
+  }
+
+  @Test
+  public void testKeyGeneration() {
+    testKeyGenerationWithDataKeySpec(null);
+    testKeyGenerationWithDataKeySpec(DataKeySpec.AES_128);
+    testKeyGenerationWithDataKeySpec(DataKeySpec.AES_256);
+  }
+
+  private void testKeyGenerationWithDataKeySpec(DataKeySpec dataKeySpec) {
+    AwsKeyManagementClient keyManagementClient = new AwsKeyManagementClient();
+    try {
+      Map<String, String> properties =
+          dataKeySpec == null
+              ? ImmutableMap.of()
+              : ImmutableMap.of(AwsProperties.KMS_DATA_KEY_SPEC, dataKeySpec.name());
+      keyManagementClient.initialize(properties);
+      KeyManagementClient.KeyGenerationResult result = keyManagementClient.generateKey(keyId);
+
+      assertThat(keyManagementClient.unwrapKey(result.wrappedKey(), keyId)).isEqualTo(result.key());
+      assertThat(result.key().limit())
+          .isEqualTo(DataKeySpec.AES_128.equals(dataKeySpec) ? 128 / 8 : 256 / 8);
```

Comment:
Can we move this to a helper function `expectedBytes(DataKeySpec spec)`? I think it'd be easier to read that way